### PR TITLE
Adding Sysdig helm onboarding to EKS-A snow

### PR DESCRIPTION
**Pull request by manuelbcd**
*Description of changes:*
Adding Sysdig helm onboarding to EKS-A snow (In progress)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by manuelbcd** [comment]
@shapirov103 please note that I'm still working on it. I submit the PR just to make it easier to get your comments in context. Thanks!

File: eks-anywhere-snow/Addons/Partner/Sysdig/values.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  namespace: sysdig
+spec:
+  refreshInterval: 1m
+  secretStoreRef:
+    name: eksa-secret-store     #The secret store name we have just created.
+    kind: ClusterSecretStore
+  target:
+    name: sysdig-secret         # Secret name in k8s
+  data:
+  - secretKey: sysdig-accesskey # which key it's going to be stored
```
Comment (elamaran11) [inline eks-anywhere-snow/Addons/Partner/Sysdig/values.yaml:14]: Please rename this to `sysdig-token` to stay consistent.

File: eks-anywhere-snow/Addons/Partner/Sysdig/values.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    remoteRef:
+      key: sysdig-accesskey     # Our secret-name goes here
```
Comment (elamaran11) [inline eks-anywhere-snow/Addons/Partner/Sysdig/values.yaml:16]: Please rename this to `sysdig-token` to stay consistent.

File: eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        kind: HelmRepository
+        name: sysdig
+        namespace: flux-system
+      version: 1.5.46
+  interval: 30s
+  url: https://charts.sysdig.com
+  values:
+    global:
+      sysdig:
+        accessKey: ${sysdigAccessKey}
```
Comment (elamaran11) [inline eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml:22]: Did it work for your locally to pull your Secrets for `sysdig-accesskey` AWS Secrets Manager?

File: eks-anywhere-snow/Addons/Partner/Sysdig/values.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+apiVersion: external-secrets.io/v1beta1
```
Comment (elamaran11) [inline eks-anywhere-snow/Addons/Partner/Sysdig/values.yaml:1]: This file should be renamed to `external-secret.yaml` to stay consistent

File: eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    spec:
+      chart: sysdig-deploy
+      reconcileStrategy: ChartVersion
+      sourceRef:
+        kind: HelmRepository
+        name: sysdig
+        namespace: flux-system
+      version: 1.5.46
+  interval: 30s
+  url: https://charts.sysdig.com
```
Comment (elamaran11) [inline eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml:18]: Why do we need `url` here if we already HelmRepostiory pointing to that url ?

**Review by elamaran11 — COMMENTED** [review]
Thanks and great work on submitting a PR for sysdig. Appreciate you doing this. I have dropped some inline comments to code snippets. But following are overall comments :
1. Test Job is missing to functionally test `sysdig` installation. Thats a criteria to pass and approve the PR.
2. Please make sure the code snippets are completely consistent with other partner addons we have.
3. Is there a reason why this is submitted only for `snow` EKSA Deployment model? Is the installation doing to be different for vSphere, BareMetal and Rover?  Any idea when we can expect sysdig snippets for other EKS Deplyment models.

Happy to have a conversation for any questions.

File: eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml (hunk 3 above)
Comment (elamaran11) [inline eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml:22]: Did the installation work fine with this and did the installation pull `sysdig-access-key` from AWS Secrets manager store in your test?

File: eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml (hunk 5 above)
Comment (manuelbcd) [reply eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml:18]: you are right, thanks

File: eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml (hunk 3 above)
Comment (manuelbcd) [reply eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml:22]: no, it didn't
I'm learning how to handle secrets and will update the PR

Comment (manuelbcd) [reply eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml:22]: I see that EKS-A plugins rely on external stores so I can't use default k8s secrets

**Comment by manuelbcd** [comment]
@elamaran11 @shapirov103 FYI

File: eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml (hunk 5 above)
Comment (elamaran11) [reply eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml]: Sounds good.

File: eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml (hunk 3 above)
Comment (elamaran11) [reply eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml]: You can follow the samples like `botkube` or `kubecost` in common and partner addons folders in different installations for sample.

Comment (manuelbcd) [reply eks-anywhere-snow/Addons/Partner/Sysdig/sysdig.yaml]: Yeah @elamaran11 , thanks. That is exactly what I did and it is passing the test and the valudation

**Comment by manuelbcd** [comment]
I'm a bit lost on how to proceed with the testjob. The testjob.yaml sample is doing nothing but deploying a busybox container to the cluster but I'm not sure on how to test a helm addon (suggestion: it would be great to have a testjob example related with the kubecost or botkube addons)   @elamaran11

**Comment by elamaran11** [comment]
> I'm a bit lost on how to proceed with the testjob. The testjob.yaml sample is doing nothing but deploying a busybox container to the cluster but I'm not sure on how to test a helm addon (suggestion: it would be great to have a testjob example related with the kubecost or botkube addons) @elamaran11

Hi @manuelbcd The `testjob.yaml` is a sample provided to show how a test job can be created and yes we used to show the same via busybox container. So the test job is not intended to test the helm addon installation because that will be verified when the GitOps sync happens and we have daily reporting in our CICD process to check that. The main intent of test job is to functional test to the functional aspect of your ISV product, whether the ISV product is functionally working fine in the K8s cluster. If you already have any functional tests for your ISV product, please containerize that and submit as a job that can run on the target cluster. Please let me know if you have any further questions.

**Review by elamaran11 — COMMENTED** [review]
Remove the two `.DS_Store` file checkedin as part of the PR.

**Comment by shapirov103** [comment]
@manuelbcd do you mind remove .DS_Store files from the PR? I see you added them to .gitignore, but this PR is not taking advantage of it yet, so the files show up.

**Comment by elamaran11** [comment]
@manuelbcd Except for the .DS_Store file and missing Functional Test job, rest looks good on the PR.

**Comment by elamaran11** [comment]
@manuelbcd Here is a ticket for Functional Test job - https://github.com/aws-samples/eks-anywhere-addons/issues/27

**Comment by shapirov103** [comment]
@manuelbcd I observe that technical validation of SysDig deployment is failing at the moment with a few pods pending:

```
 sysdig-agent-6fthk                          ●       1/1                      0 Running        10.0.0.214        s-i-8bad74b9e13b35e5c        42m         │
│ sysdig-agent-9x2gm                          ●       0/0                      0 Pending        n/a               n/a                          42m         │
│ sysdig-agent-bgvq6                          ●       0/0                      0 Pending        n/a               n/a                          42m         │
│ sysdig-agent-mm9gg                          ●       1/1                      0 Running        10.0.0.101        s-i-88e94fa614f36e9b2        42m         │
│ sysdig-agent-p8zhh                          ●       0/0                      0 Pending        n/a               n/a                          42m         │
│ sysdig-agent-q8rxd                          ●       1/1                      0 Running        10.0.0.81         s-i-8b086ffa3525e0543        42m         │
│ sysdig-kspmcollector-759d74589f-zl4bq       ●       1/1                      0 Running        10.1.4.52         s-i-8b086ffa3525e0543        42m         │
│ sysdig-node-analyzer-4ms54                  ●       3/3                      0 Running        10.0.0.81         s-i-8b086ffa3525e0543        42m         │
│ sysdig-node-analyzer-b82vg                  ●       3/3                      0 Running        10.0.0.43         s-i-84960b24e1180cd93        42m         │
│ sysdig-node-analyzer-jxcg2                  ●       3/3                      0 Running        10.0.0.139        s-i-853e1770cd5d27696        42m         │
│ sysdig-node-analyzer-khmjq                  ●       3/3                      0 Running        10.0.0.101        s-i-88e94fa614f36e9b2        42m         │
│ sysdig-node-analyzer-n7p4n                  ●       0/0                      0 Pending        n/a               n/a                          42m         │
│ sysdig-node-analyzer-r7qdj                  ●       3/3                      0 Running        10.0.0.79         s-i-8403823df04217822        42m
```

I am unclear on the actual reason, but potentially the issue could be addressed by limiting the daemonset to worker nodes only. Worker nodes are marked with the following label on snow: `cluster.x-k8s.io/owner-kind: MachineSet`.

The GitOps PR could be modified to use that as a selector potentially.

**Comment by manuelbcd** [comment]
Removed agent tolerations from the helm parameters to avoid the execution of the agent in control-plane nodes

**Review by elamaran11 — APPROVED** [review]
LGTM

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**